src/utils/positionUtils.ts

```diff
@@ -389,34 +426,41 @@ export const calcTradeFee = ({
   const takerProportionalFee = Big6Math.div(Big6Math.mul(notional, takerProportionalFeeNumerator), takerFee.scale)
 
   const takerLinearFee = Big6Math.mul(notional, takerFee.linearFee)
-  const subtractiveFee = Big6Math.mul(takerLinearFee, referralFee)
+  const notionalTakerFee = Big6Math.mul(marketTakerFee, notional)
 
-  const marketFee = Big6Math.mul(takerLinearFee - subtractiveFee, marketTakerFee)
-  const tradeFee = subtractiveFee + marketFee
+  const tradeFee = notionalTakerFee + takerProportionalFee
+  const referralFeeNotional = Big6Math.mul(tradeFee, referralFee)
   const feeBasisPoints = !Big6Math.isZero(tradeFee) ? Big6Math.div(tradeFee, notional) : 0n
-  const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee - tradeFee
+  // TODO: Check - Old impact calc subtracts trade fee
+  // const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee - tradeFee
+  const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee
+  const priceImpact = positionDelta !== 0n ? Big6Math.div(tradeImpact, Big6Math.abs(positionDelta)) : 0n
+  const priceImpactPct = priceImpact !== 0n ? Big6Math.div(priceImpact, Big6Math.abs(notional)) : 0n
 
   tradeFeeInfo = {
-    tradeFee,
-    tradeImpact,
-    feeBasisPoints,
-    proportionalFee: takerProportionalFee,
-    linearFee: takerLinearFee,
-    adiabaticFee: takerAdiabaticFee,
+    tradeFee: {
+      total: tradeFee,
+      basisPoints: feeBasisPoints,
+      components: {
+        referralFee: referralFeeNotional,
+      },
+    },
+    tradeImpact: {
+      total: tradeImpact,
+      components: {
+        proportionalFee: takerProportionalFee,
+        linearFee: takerLinearFee,
+        adiabaticFee: takerAdiabaticFee,
+      },
+    },
+    priceImpact: {
```

Comment:
Having separate price impact and trade impact keys might be confusing. I think we can do `tradeImpact.impactPerPosition = priceImpact` and `tradeImpact.basisPoints = priceImpactPct`